File: python/128_Longest_Consecutive_Sequence.cs

public class Solution {
    public int LongestConsecutive(int[] nums) {

        HashSet<int> setOfNums = new HashSet<int>(nums);
        int longestSoFar = 0;

        foreach(int x in setOfNums) 
        {
            if(!setOfNums.Contains(x-1)){
                int len = 0;
                while(setOfNums.Contains(x + len))
                    len++;
                longestSoFar = Math.Max(longestSoFar, len);
            }
        }
        return longestSoFar; 
    }
}

File: python/128_Longest_Consecutive_Sequence.py


def longestConsecutive(nums):
    non_repeated_characters=set(nums)
    max_len=0

    for x in nums:
        if not x-1 in non_repeated_characters:
            curr_len=0
            while x+curr_len in non_repeated_characters:
                curr_len+=1
            
            max_len=max(max_len, curr_len) 
    
    return max_len